src/clr-angular/data/datagrid/render/header-renderer.spec.ts

```diff
@@ -76,15 +78,24 @@ export default function(): void {
         StateDebouncer,
         TableSizeService,
         Renderer2,
+        ColumnsService,
       ]);
       domAdapter = <MockDomAdapter>context.getClarityProvider(DomAdapter);
       organizer = <MockDatagridRenderOrganizer>context.getClarityProvider(DatagridRenderOrganizer);
-      stateSub = context.clarityDirective.columnState;
+      columnsService = context.getClarityProvider(ColumnsService);
+      context.clarityDirective.setColumnState(0);
+      stateSub = columnsService.columns[0];
+    });
+
+    it('sets its state in columns service at given index', function() {
+      expect(columnsService.columns.length).toBe(1);
```

Comment:
This test isn't checking that the state is actually set properly, just that there is a value.

Reply:
But every property will be undefined here as this is just testing whether the header is sending its initial pristine state to ColumnService.